Automattic/newspack-plugin
# Handoff Component

### All Submissions:






### Changes proposed in this Pull Request:

This PR introduces the `Handoff` component, which is the UI and functionality to allow linking out to other plugin dashboards. Here are the elements of this functionality:

1) Renders a `Button`
2) The `Button` initially shows a loading state as information on the plugin is retrieved via API call. (Alternatively all plugin info could be set as props, avoiding the delay)
3) If any the plugin is not installed/activated the `Button` will be disabled and the text will read "{plugin-name} not installed"
4) When clicked, a `Modal` is shown to prepare the user for the handoff.
5) When the primary button in the modal is clicked, the user is redirected to the plugin's dashboard.
6) From this point forward, a Newsspack button w/logo will be shown in the admin bar. Clicking takes the user back to the last Newspack dashbaord URL they visited.

Properties:
`plugin`: Slug of the plugin for the handoff. 
`modalBody`: Body text for the modal. Default: ''
`modalTitle`: Title for the modal. Default: "Manage {plugin-name}"
`primaryButton`: Text for the primary button. Default: "Manage {plugin-name}"
`primaryModalButton`: Text for primary modal button. Default: "Manage"
`dismissModalButton`: Text for secondary modal button that dismisses it. Default: "Dismiss"



### How to test the changes in this Pull Request:

1. Checkout the branch, then run `npm ci && npm run clean && npm run build:webpack`
2. Navigate to `Newspack->Components demo` (`/wp-admin/admin.php?page=newspack-components-demo`)
3. Observe Handoff Buttons for AMP, Jetpack, Site Kit, and WooCommerce.
4. If these are installed and active, clicking the link should open the modal. 
5. Click `Manage` button in modal, and verify redirection to the plugin's dashboard page. 
6. Verify Newspack link in the admin bar.
7. Click Newspack link in the admin bar. Verify return to Newspack. Verify Newspack link in admin bar is gone.

### Other information:
